docs/cookbook/getBusinessOpenStateText.mjs

```diff
@@ -52,17 +53,18 @@ function getBusinessOpenStateText(now, timeZone, businessHours, soonWindow) {
     const { open, close } = todayHours;
     const todayOrTomorrow = Temporal.Time.compare(close, open) >= 0 ? today : tomorrow;
     businessHoursOverlappingToday.push({
-      open: today.toDateTime(open).toInstant(timeZone),
-      close: todayOrTomorrow.toDateTime(close).toInstant(timeZone)
+      open: now.timeZone.getInstantFor(today.toDateTime(open)),
```

Comment:
I'l do a full review later tonight (tomorrow at latest), but first a quick observation: I don't think we want to expose TimeZone methods as the recommended pattern for use cases like this. Instead, I think it's generally better to use a chainable API instead, e.g. 

```js
today.toDateTime(open).toZonedDateTime(now.timeZone).toInstant()
``` 

This is a little longer but probably gives users a better sense of the patterns we expect to see in the wild, and may help users better understand how these types relate to each other. 

Also, I'm not sure that `toInstant()` is needed at the end. Seems cleaner to set `open` and `close` to be ZonedDateTime instances and then change `inRange` to use `ZonedDateTIme.compare` instead.

Reply:
Agreed, this example needs to be rewritten to use the full capabilities of ZonedDateTime as you did in #700, but that requires add() and with() which are currently unimplemented in the polyfill. This is only a stopgap to keep the example working.

Reply:
OK sounds good.